File: 2015/02/main.ts
#!/usr/bin/env -S deno run --allow-read

const inputText = await Deno.readTextFile(
  new URL("input.txt", import.meta.url),
);

let answerOne = 0;
let answerTwo = 0;

for (const line of inputText.split("\n")) {
  const match = line.match(/^(\d+)x(\d+)x(\d+)$/);
  if (!match) continue;
  const [, L, W, H] = match.map((n) => Number.parseInt(n));
  const sides = [
    2 * (L * W),
    2 * (W * H),
    2 * (H * L),
  ];
  const dimensions = [L, W, H].sort((a, b) => (a - b));
  sides.forEach((side) => (answerOne += side));
  answerOne += Math.min(...sides) / 2;
  answerTwo += (dimensions[0] * 2) + (dimensions[1] * 2);
  answerTwo += L * W * H;
}

console.log(`Answer one: ${answerOne}`);
console.log(`Answer two: ${answerTwo}`);

File: 2015/02/main.zig
const std = @import("std");
const assert = std.debug.assert;
const parseInt = std.fmt.parseInt;

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    const file = try std.fs.cwd().openFile("2015/02/input.txt", .{});
    defer file.close();

    const reader = file.reader();
    var buffer: [1024]u8 = undefined;

    var answer_one: i32 = 0;
    var answer_two: i32 = 0;

    while (try reader.readUntilDelimiterOrEof(&buffer, '\n')) |line| {
        var split = std.mem.splitScalar(u8, line, 'x');
        const L = try parseInt(i32, split.next().?, 10);
        const W = try parseInt(i32, split.next().?, 10);
        const H = try parseInt(i32, split.next().?, 10);

        var dimensions: [3]i32 = .{ L, W, H };
        std.mem.sort(i32, &dimensions, {}, std.sort.asc(i32));

        const sides: [3]i32 = .{ 2 * (L * W), 2 * (W * H), 2 * (H * L) };
        var min_side: i32 = sides[0];
        for (sides) |side| {
            answer_one += side;
            if (side < min_side) min_side = side;
        }

        answer_one += @divFloor(min_side, 2);
        answer_two += (dimensions[0] * 2) + (dimensions[1] * 2);
        answer_two += L * W * H;
    }

    try stdout.print("Answer 1: {d}\n", .{answer_one});
    try stdout.print("Answer 2: {d}\n", .{answer_two});
}
